dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -175,6 +169,55 @@ def check_nd_call_func(
     )
 
 
+def _make_unary_func(
+    name, dpt_unary_fn, fn_docstring, mkl_fn_to_call=None, mkl_impl_fn=None
+):
+    impl_fn = dpt_unary_fn.get_implementation_function()
+    type_resolver_fn = dpt_unary_fn.get_type_result_resolver_function()
+
+    def _call_func(src, dst, sycl_queue, depends=None):
+        """A callback to register in UnaryElementwiseFunc class of dpctl.tensor"""
+
+        if depends is None:
+            depends = []
+
+        if mkl_fn_to_call is not None and mkl_fn_to_call(sycl_queue, src, dst):
+            # call pybind11 extension for unary function from OneMKL VM
+            return mkl_impl_fn(sycl_queue, src, dst, depends)
+        return impl_fn(src, dst, sycl_queue, depends)
+
+    func = dpt_unary_fn.__class__(
+        name, type_resolver_fn, _call_func, fn_docstring
+    )
+    return func
+
+
+def _make_binary_func(
+    name, dpt_binary_fn, fn_docstring, mkl_fn_to_call=None, mkl_impl_fn=None
+):
+    impl_fn = dpt_binary_fn.get_implementation_function()
+    type_resolver_fn = dpt_binary_fn.get_type_result_resolver_function()
+    fn_inplce = dpt_binary_fn.get_implementation_inplace_function()
```

Comment:
```suggestion
    impl_fn = dpt_binary_fn.get_implementation_function()
    type_resolver_fn = dpt_binary_fn.get_type_result_resolver_function()
    fn_inplce = dpt_binary_fn.get_implementation_inplace_function()
```

You also need to read `acceptance_fn = dpt_binary_fn.get_type_promotion_path_acceptance_function()` and pass it to the costructor. This is needed for the `dpnp.divide` function to properly return a float32/float64 for `int8`/`int8`.